packages/react-components/react-virtualizer/src/hooks/useVirtualizerMeasure.ts

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import { canUseDOM } from '@fluentui/react-utilities';
+import { IVirtualizerMeasureDynamicProps, IVirtualizerMeasureProps } from './useVirtualizerMeasure.types';
+import { debounce } from '../utilities/debounce';
+import * as ReactDOM from 'react-dom';
+
+/**
+ * React hook that measures virtualized space based on a static size to ensure optimized virtualization length.
+ */
+export const useStaticVirtualizerMeasure = (
+  virtualizerProps: IVirtualizerMeasureProps,
+): {
+  virtualizerLength: number;
+  bufferItems: number;
+  bufferSize: number;
+  useScrollRef: (ref: React.MutableRefObject<HTMLElement | null>) => void;
+} => {
+  const { defaultItemSize, direction = 'vertical' } = virtualizerProps;
+
+  const [virtualizerLength, setVirtualizerLength] = React.useState(0);
+  const [virtualizerBufferItems, setVirtualizerBufferItems] = React.useState(0);
+  const [virtualizerBufferSize, setVirtualizerBufferSize] = React.useState(0);
+
+  // The ref the user sets on their scrollView - Defaults to document.body to ensure no null on init
+  const container: React.MutableRefObject<HTMLElement | null> = React.useRef<HTMLElement | null>(null);
+
+  const useVirtualizerScrollRef = (ref: React.MutableRefObject<HTMLElement | HTMLDivElement | null>) => {
+    React.useEffect(() => {
+      if (ref.current !== container.current) {
+        if (container.current) {
+          resizeObserver?.unobserve(container.current);
+        }
+
+        // Update
+        container.current = ref.current;
+
+        // Only observe if not null
+        if (container.current) {
+          resizeObserver?.observe(container.current);
+        }
+      }
+    });
+  };
+
+  // the handler for resize observer
+  const handleResize = debounce(
+    React.useCallback(() => {
+      if (!container.current) {
+        return;
+      }
+
+      const containerSize =
+        direction === 'vertical'
+          ? container.current.getBoundingClientRect().height
+          : container.current.getBoundingClientRect().width;
+
+      /*
+       * Number of items required to cover viewport.
+       */
+      const length = Math.ceil(containerSize / defaultItemSize + 1);
+
+      /*
+       * Number of items to append at each end, i.e. 'preload' each side before entering view.
+       */
+      const bufferItems = Math.max(Math.floor(length / 4), 2);
```

Comment:
why did you settle on a quarter of the items for each buffer?

Reply:
One fourth was considered to be a good 'overflow' buffer for the intersection observers to exist in, we can tweak this result in the future but 1/4th scales well at both high/low length values (generally higher length = larger viewport or smaller items; where scrolling is likely to pass over more overflow items between intersection calls).